src/lib/register-user.js

```diff
@@ -0,0 +1,34 @@
+import {
+  getCredentialRegister,
+  validateRegister,
+  register,
+  USERNAME_KEY,
+  CONFIRM_KEY,
+  PASS_KEY,
+} from "./auth.js";
+
+document.addEventListener("DOMContentLoaded", () => {
+  const registerForm = document.getElementById("registerForm");
+
+  registerForm.addEventListener("submit", (e) => {
+    e.preventDefault();
+    const credential = getCredentialRegister();
+    const validate = validateRegister(credential);
+
+    if (validate) {
+      if (validate[USERNAME_KEY]) {
+        document.getElementById("errorUser").innerHTML = validate[USERNAME_KEY];
+      }
+      if (validate[PASS_KEY]) {
+        document.getElementById("errorPass").innerHTML = validate[PASS_KEY];
+      }
+      if (validate[CONFIRM_KEY]) {
+        document.getElementById("errorConfi").innerHTML = validate[CONFIRM_KEY];
+      }
+    }
```

Comment:
don't use innerHTML, use document.getElementById("errorUser").textContent = validate[USERNAME_KEY].
using innerHTML is considered dangerous nowadays 
see --> https://www.dhairyashah.dev/posts/why-innerhtml-is-a-bad-idea-and-how-to-avoid-it/